pilot/pkg/config/monitor/file_snapshotter.go

```diff
@@ -0,0 +1,129 @@
+// Copyright 2018 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package monitor
+
+import (
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"sort"
+
+	"istio.io/istio/pilot/pkg/config/kube/crd"
+	"istio.io/istio/pilot/pkg/model"
+	"istio.io/istio/pkg/log"
+)
+
+var (
+	supportedExtensions = map[string]bool{
+		".yaml": true,
+		".yml":  true,
+	}
+)
+
+// FileSnapshotter holds a reference to a file directory that contains crd
+// config and filter criteria for which of those configs will be parsed.
+type FileSnapshotter struct {
+	root             string
+	configTypeFilter map[string]bool
+}
+
+// NewFileSnapshotter returns a snapshotter.
+// If no types are provided in the descriptor, all IstioConfigTypes will be allowed.
+func NewFileSnapshotter(root string, descriptor model.ConfigDescriptor) *FileSnapshotter {
+	snapshotter := &FileSnapshotter{
+		root:             root,
+		configTypeFilter: make(map[string]bool),
+	}
+
+	types := descriptor.Types()
+	if len(types) == 0 {
+		types = model.IstioConfigTypes.Types()
+	}
+
+	for _, k := range types {
```

Comment:
Some comment would be good - why not read all types ?

Reply:
This was ported over from the original code that handled which config types were valid. It does use all types unless a filter of types aka `configdescriptor` is provided. I could remove this functionality if we just want to use all types all the time but left the behavior as is in the refactor.